components/recurring-contributions/RecurringContributionsPopUp.js

```diff
@@ -180,12 +241,13 @@ const RecurringContributionsPopUp = ({ contribution, status, onCloseEdit, accoun
           <Flex flexGrow={1 / 4} width={1} alignItems="center" justifyContent="center">
             <StyledButton
               buttonSize="tiny"
+              buttonStyle="secondary"
               minWidth={75}
               loading={loadingCancellation}
               data-cy="recurring-contribution-cancel-yes"
               onClick={async () => {
                 try {
-                  await submitCancellation({ variables: { order: { id: contribution.id } } });
+                  await submitCancellation({ variables: { order: { id: contribution.id }, reason: textAreaState } });
```

Comment:
The API will receive the internationalized version of the string, which is not ideal. Our goal with this is to surface it in an admin dashboard with something like:

```sql
SELECT "data" ->> 'reason', COUNT(id)
FROM "Activities"
WHERE "type" = 'subscription.canceled'
GROUP BY "data" ->> 'reason'
```

If we passed the international version to the API, we'll end up with something like:

| reason | count |
| --------- | --- |
| No longer want to back the collective | 3 |
| Je ne veux plus soutenir le collectif | 1 |
| Ya no quieren respaldar al colectivo | 1 |
| A custom message entered through other | 1 |

It would be much more interesting to have:

| reason | count |
| --------- | --- |
| NO_LONGER_WANT_TO_SUPPORT | 5 |
| A custom message entered through other | 1 |

Reply:
Makes sense.  I stopped internationalizing this data, but also I thought about it a bit more and realized that for analytics you might want to be able to see both when "Other" is chosen, as well as what the actual reason is, so I added something called a `reasonCode` which are the keys you suggested in the `i18nReason` object, and the `reason` is the same now, except for if somebody was displayed a textarea, in which case it is the content that was there.  Felt like the best way to capture all data, but let me know if you disagree.